featurebyte/service/feature_store_warehouse.py

```diff
@@ -192,10 +105,10 @@ async def list_schemas(
         db_session = await self.session_manager_service.get_feature_store_session(
             feature_store=feature_store, get_credential=get_credential
         )
-        # check database exists
-        await self.check_database_exists(db_session=db_session, database_name=database_name)
-
-        return await db_session.list_schemas(database_name=database_name)
+        try:
+            return await db_session.list_schemas(database_name=database_name)
+        except Exception as exc:
+            raise DatabaseNotFoundError(f"Database {database_name} not found.") from exc
```

Comment:
Could this fail because of other reasons such as connection issue? Thinking whether it would be useful to include some information e.g. the error message from `exc` in the raised error.

Reply:
@YS-L Switched to using the database specific error when no schema is found